ConsoleCalculator/Calculator.cs

```diff
@@ -1,14 +1,139 @@
 ﻿using System;
+using System.Collections.Generic;
 
 namespace ConsoleCalculator
 {
     public class Calculator
-    {
-
-        public string SendKeyPress(char key)
-        {
-            // Add your implementation here.
-            throw new NotImplementedException();
+    {   
+        public const string Numbers = "0123456789.";
+        public const string Sign = "sS";
+        public const string Operators = "+-xX/";
+        public const string EqualsSign = "=";
+        public const string ClearScreen = "cC";
+
+        public enum Events { NumberFeed, SignFeed, OperatorFeed, DisplayResult, ClearScreen, Error, Default }
+        
+        private OperandState result;
+        private OperandState operand;
+        private OperandState currentOperandDisplayed;
+        private char? previousOperator = null;
+
+        public Calculator() {
+            // Display 0 initially
+            Console.WriteLine("0");
+
+            result = new OperandState();
+            operand = new OperandState();
+
+            currentOperandDisplayed = result;
+        }
+        public string SendKeyPress(char key)
+        {   
+            Events _event = Events.Default;
+
+            if (Numbers.Contains(key.ToString())) 
+                _event = Events.NumberFeed;
+            else if (Operators.Contains(key.ToString())) 
+                _event = Events.OperatorFeed;
+            else if (Sign.Contains(key.ToString())) 
+                _event = Events.SignFeed;
+            else if (EqualsSign.Contains(key.ToString())) 
+                _event = Events.DisplayResult;
+            else if (ClearScreen.Contains(key.ToString())) 
+                _event = Events.ClearScreen;
+            else _event = Events.Error;
+
+            return FeedToFSM(_event, key);
+        }
+
+        public string FeedToFSM(Events type, char key) {
+            switch (type) {
+                case Events.NumberFeed:
+                    return currentOperandDisplayed.AddAndReturn(key);
+                
+                case Events.SignFeed:
+                    return currentOperandDisplayed.ToggleSignAndReturn();
+                
+                case Events.OperatorFeed:
+                    this.SwitchNumbersBeingDisplayed();
+                    return EvaluatePreviousExpression(key);
+                
+                case Events.DisplayResult:
+                    return EvaluatePreviousExpression(null);
+                
+                case Events.ClearScreen:
+                    this.Restore("0", "", null);
+                    return result.content;
+                
+                case Events.Default:
+                default:
+                    return currentOperandDisplayed.content;
+            }
+        }
+
+        public string EvaluatePreviousExpression(char? currentOperator) {
+            ConvertNumbersInCorrectFormat();
+
+            // No pending operations
+            if (previousOperator == null) {

```

Comment:
Not getting the intent of this code here.

Reply:
Take equation to be '4+2-3='. Now, when '+' is encountered, the previous equation is just a number. So, we just assign 'previousOp = +' and return 4 to be displayed. But when '-' is encountered, at that time, we know that 'number2' has been fully given, so it is time to evaluate '4+2' and return that result. Now, the state of the system is '6-'. And continued so on.